core/src/cosmos/bech32/CosmosLikeBech32ParametersHelpers.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ * CosmosLikeBech32ParametersHelpers
+ *
+ * Created by Gerry Agbobada on 2020/02/10
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#pragma once
+
+#include <api/ErrorCode.hpp>
+#include <collections/strings.hpp>
+#include <math/BigInt.h>
+#include <math/bech32/Bech32Parameters.h>
+#include <utils/Exception.hpp>
+#include <utils/hex.h>
+#include <api/CosmosBech32Type.hpp>
+
+
+namespace ledger {
+namespace core {
+namespace cosmos {
+using namespace Bech32Parameters;
+static const Bech32Parameters::Bech32Struct getBech32Params(api::CosmosBech32Type type) {
+    static const Bech32Struct COSMOS_PUB = {
+        "cosmospub",
+        "cosmospub",
+        "1",
+        6,
+        {0x3b6a57b2ULL, 0x26508e6dULL, 0x1ea119faULL, 0x3d4233ddULL, 0x2a1462b3ULL},
+        {0x00},
+        {0x00}};
+    static const Bech32Parameters::Bech32Struct COSMOS_PUB_VAL = {
+        "cosmosvaloperpub",
+        "cosmosvaloperpub",
+        "1",
+        6,
+        {0x3b6a57b2ULL, 0x26508e6dULL, 0x1ea119faULL, 0x3d4233ddULL, 0x2a1462b3ULL},
+        {0x00},
+        {0x00}};
+    static const Bech32Parameters::Bech32Struct COSMOS = {
+        "cosmos",
+        "cosmos",
+        "1",
+        6,
+        {0x3b6a57b2ULL, 0x26508e6dULL, 0x1ea119faULL, 0x3d4233ddULL, 0x2a1462b3ULL},
+        {0x01},
+        {0x01}};
+    static const Bech32Parameters::Bech32Struct COSMOS_VAL = {
+        "cosmosvaloper",
+        "cosmosvaloper",
+        "1",
+        6,
+        {0x3b6a57b2ULL, 0x26508e6dULL, 0x1ea119faULL, 0x3d4233ddULL, 0x2a1462b3ULL},
+        {0x01},
+        {0x01}};
```

Comment:
It won't be trivial to make these constexpr, as there are strings and vectors in the definition : 
```
            struct Bech32Struct {
                std::string name;
                std::string hrp;
                std::string separator;
                size_t checksumSize;
                std::vector<unsigned long long> generator;
                std::vector<uint8_t> P2WPKHVersion;
                std::vector<uint8_t> P2WSHVersion;
[...]
```
Changing this would need modifications in the bitcoin code that also uses the Bech32Struct type for segwit support, so I won't do that just now, but that's definitely something we can improve !